store/postgres/src/notification_listener.rs

```diff
@@ -245,8 +245,20 @@ impl JsonNotification {
                         "SELECT payload FROM large_notifications WHERE id = $1",
                         &[&(payload_id as i32)],
                     )
-                    .map_err(|_| format_err!("No payload found for notification {}", payload_id))?;
-
+                    .map_err(|e| {
+                        format_err!(
+                            "Error retrieving payload for notification {}: {}",
+                            payload_id,
+                            e
+                        )
+                    })?;
+
+                if payload_rows.is_empty() || payload_rows.get(0).is_empty() {
```

Comment:
In what case would `payload_rows.get(0).is_empty()` be true?

Reply:
I thought the same when I wrote that: I don't think it could ever happen, but out of an overabundance of caution, I covered that case, too.